### PR TITLE
[AIRFLOW-4113] Unpin boto3

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4113

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Right now we're pinning boto on <1.8 because of an underlying issue in moto. This issue has been solved, and therefore it would be nice to update again to a more recent version of boto.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by kaxil** [comment]
CI Passed: https://travis-ci.com/astronomer/airflow/builds/142182607?

File: setup.py [hunk 1]
```diff
@@ -380,7 +380,7 @@ def write_version(filename: str = os.path.join(*["airflow", "git_version"])):
     'ipdb',
     'jira',
     'mongomock',
-    'moto==1.3.5',
+    'moto>=1.3.14',
```
Comment (ashb) [inline setup.py]: Something to disallow Moto 2?

**Review by ashb — CHANGES_REQUESTED** [review]
One small change

File: setup.py (hunk 1 above)
Comment (kaxil) [reply setup.py]: Updated

**Review by ashb — APPROVED** [review]

**Pull request merged**